wire-library/wire-moshi-adapter/src/main/java/com/squareup/wire/WireJsonAdapterFactory.kt

```diff
@@ -197,6 +264,64 @@ class WireJsonAdapterFactory private constructor(
       }
     }.nullSafe()
 
+    /**
+     * Moshi doesn't know enough to follow a `@Uint64String List<Long>` wants to be treated as a
+     * `List<@Uint64String Long>` and so we have to do it manually.
+     *
+     * TODO delete when Moshi can handle that; see
+     * [moshi/issues/666](https://github.com/square/moshi/issues/666)
+     */
+    internal val LIST_OF_UINT64_STRING_JSON_ADAPTER = object : JsonAdapter<List<Long>>() {
```

Comment:
We should build a thing to take a `@Qualified List<T>` and turn it into a `List<@Qualified T>` for Wire to use

Reply:
I thought this wasn't doable in Java8/Kotlin ?

Reply:
It's not doable on Android. Perfectly doable on the JVM though for both Java 8 or Kotlin.

Reply:
I was thinking something way way dumber. Just a general class that wraps an `Adapter<T>` as a List so we don’t need to do it 2x